security/WFCORE-3873_SNI_Support.adoc

```diff
@@ -0,0 +1,76 @@
+= SNI Support
+:author:            Stuart Douglas
+:email:             sdouglas@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+This aim of this issue is to provide SNI support for all SSL use cases that use SSLEngine.
+
+This will allow for users to specify match rules for incoming SNI names, and at runtime
+the appropriate SSL context will be selected for the name.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/WFCORE-3873[WFCORE-3873]
+
+=== Related Issues
+
+* https://issues.jboss.org/browse/EAP7-726
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+=== Affected Projects or Components
+
+* Wildfly Core
+* Undertow
+* Elytron
+
+=== Other Interested Projects
+
+== Requirements
+
+=== Hard Requirements
+
+This will involve adding a new `server-sni-ssl-context` resource to the Elytron subsystem. This resource will have
+a non-optional attribute named `default-ssl-context` that references an existing server-ssl-context.
+
+In addition this resource can have any number of sni-match subresources. These subresources map names (which may be
+wildcard names) to `server-ssl-context` resources, and will be used by the SNI support to determine the correct context
+to choose. Wildcard names use `*` as a wildcard, and can only be used to match a single level of subdomain in much the
+same way as with wildcard certificates.
```

Comment:
Out of curiosity - is something like `*.*.example.com` or just `*.*.com` possible? I guess we want to support only one wildcard in the `sni-match`, correct?

Reply:
@stuartwdouglas I've just realized this question was not answered. I still suppose the right behavior is not to allow this.

Reply:
Yea, we only want to allow a single wildcard